platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/extensions/MapExtensions.kt

```diff
@@ -20,3 +20,14 @@ package org.gradle.configurationcache.extensions
 internal
 fun <V> Map<String, V>.filterKeysByPrefix(prefix: String): Map<String, V?> =
     filterKeys { key -> key.length > prefix.length && key.startsWith(prefix) }
+
+
+/**
+ * Returns a map with the values mapped to the keys.
+ */
+internal
+fun <K, V> Map<K, V>.invert() = HashMap<V, K>(size).also { result ->
```

Comment:
🤔 How this function will behave in case of duplicated values in original map? Apparently, it'll lose the data.
Given that it placed in common `MapExtensions` file and thus able to be used in many places, I would suggest at least to note it in the doc of the function